datachannel.go

```diff
@@ -38,6 +38,12 @@ type DataChannel struct {
 	Config
 	stream *sctp.Stream
 	log    logging.LeveledLogger
+
+	// stats
```

Comment:
Do these all need to be thread safe?

Reply:
I made them thread safe! I was thinking this will be called inside webrtc.DataChannel but I forget about DetachDataChannel().

Thanks for the catch!